src/modules/aws-ec2/iam-instance-profile/removeIamInstanceProfile.ts

```diff
@@ -0,0 +1,24 @@
+import { EC2 } from "aws-sdk";
+import { logger } from "../../../logger";
+import { getIamInstanceProfileAssociations } from "./getIamInstanceProfileAssociations";
+
+export const removeIamInstanceProfile = async (instanceId: string) => {
+  const ec2 = new EC2();
+  try {
+    const responseData = await getIamInstanceProfileAssociations(instanceId);
+    if (!responseData.IamInstanceProfileAssociations) {
+      return;
+    }
+    const profileAssociations = responseData.IamInstanceProfileAssociations;
+    profileAssociations.forEach(async association => {
```

Comment:
The `forEach` function on the array prototype is not async aware. Each callback will be processed one after the other without waiting for the previous, if this is desired consider looping over them in a way that will allow the usage of async/await, for example a for of loop.

If you want these executions to be ran in 'parallel' still, consider mapping over the array instead and using `Promise.all()` on the resulting array of promises.